metals/src/main/scala/scala/meta/internal/parsing/ClassFinder.scala

```diff
@@ -5,100 +5,153 @@ import scala.collection.mutable
 
 import scala.meta.Defn
 import scala.meta.Member
+import scala.meta.Mod
 import scala.meta.Pkg
 import scala.meta.Position
 import scala.meta.Self
 import scala.meta.Tree
+import scala.meta.Type
 import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.parsing.finder.ClassKind
+import scala.meta.internal.parsing.finder.MangledClassName
+import scala.meta.internal.parsing.finder.ShortClassName
 import scala.meta.io.AbsolutePath
 
 import org.eclipse.{lsp4j => l}
 
-final case class ClassWithPos(
-    path: String,
-    friendlyName: String,
-    description: String,
-)
+final case class ClassArtifact(
+    kind: ClassKind,
+    mangledClassName: MangledClassName,
+    shortClassName: ShortClassName,
+    extensionSuffix: String,
```

Comment:
after some time, I had no idea what `friendlyName` and `description` are. `MangledClassName` and `ShortClassName` have defined meaning which makes it easier to see what's going on